diff -Naur linux-2.6.37-rc2/drivers/tty/tty_io.c linux-2.6.37-rc2.new/drivers/tty/tty_io.c
--- linux-2.6.37-rc2/drivers/tty/tty_io.c	2010-11-18 18:58:47.000000000 +0100
+++ linux-2.6.37-rc2.new/drivers/tty/tty_io.c	2010-11-18 19:21:30.000000000 +0100
@@ -187,6 +187,7 @@
 	if (tty->dev)
 		put_device(tty->dev);
 	kfree(tty->write_buf);
+	tty_sched_destroy_group(tty);
 	tty_buffer_free_all(tty);
 	kfree(tty);
 }
@@ -2868,6 +2869,7 @@
 	tty->index = idx;
 	tty_line_name(driver, idx, tty->name);
 	tty->dev = tty_get_device(tty);
+	tty_sched_create_group(tty);
 }
 
 /**
diff -Naur linux-2.6.37-rc2/include/linux/sched.h linux-2.6.37-rc2.new/include/linux/sched.h
--- linux-2.6.37-rc2/include/linux/sched.h	2010-11-18 18:58:48.000000000 +0100
+++ linux-2.6.37-rc2.new/include/linux/sched.h	2010-11-18 19:00:55.000000000 +0100
@@ -1930,6 +1930,20 @@
 
 extern unsigned int sysctl_sched_compat_yield;
 
+#ifdef CONFIG_SCHED_DESKTOP
+int sched_tty_sched_handler(struct ctl_table *table, int write,
+		void __user *buffer, size_t *lenp,
+		loff_t *ppos);
+
+extern unsigned int sysctl_sched_tty_sched_enabled;
+
+void tty_sched_create_group(struct tty_struct *tty);
+void tty_sched_destroy_group(struct tty_struct *tty);
+#else
+static inline void tty_sched_create_group(struct tty_struct *tty) { }
+static inline void tty_sched_destroy_group(struct tty_struct *tty) { }
+#endif
+
 #ifdef CONFIG_RT_MUTEXES
 extern int rt_mutex_getprio(struct task_struct *p);
 extern void rt_mutex_setprio(struct task_struct *p, int prio);
diff -Naur linux-2.6.37-rc2/include/linux/tty.h linux-2.6.37-rc2.new/include/linux/tty.h
--- linux-2.6.37-rc2/include/linux/tty.h	2010-11-18 18:58:48.000000000 +0100
+++ linux-2.6.37-rc2.new/include/linux/tty.h	2010-11-18 19:00:55.000000000 +0100
@@ -329,6 +329,9 @@
 	/* If the tty has a pending do_SAK, queue it here - akpm */
 	struct work_struct SAK_work;
 	struct tty_port *port;
+#ifdef CONFIG_SCHED_DESKTOP
+	struct task_group *tg;
+#endif
 };
 
 /* Each of a tty's open files has private_data pointing to tty_file_private */
diff -Naur linux-2.6.37-rc2/init/Kconfig linux-2.6.37-rc2.new/init/Kconfig
--- linux-2.6.37-rc2/init/Kconfig	2010-11-18 18:58:48.000000000 +0100
+++ linux-2.6.37-rc2.new/init/Kconfig	2010-11-18 19:00:55.000000000 +0100
@@ -675,6 +675,19 @@
 
 endif # CGROUPS
 
+config SCHED_DESKTOP
+	bool "Desktop centric group scheduling"
+	depends on EXPERIMENTAL
+	select CGROUPS
+	select CGROUP_SCHED
+	select FAIR_GROUP_SCHED
+	select RT_GROUP_SCHED
+	select BLK_CGROUP
+	help
+	  This option optimizes the group scheduler for common desktop workloads,
+	  by creating separate per tty groups. This separation of workloads isolates
+	  aggressive CPU burners (like build jobs) from desktop applications.
+
 menuconfig NAMESPACES
 	bool "Namespaces support" if EMBEDDED
 	default !EMBEDDED
diff -Naur linux-2.6.37-rc2/kernel/sched.c linux-2.6.37-rc2.new/kernel/sched.c
--- linux-2.6.37-rc2/kernel/sched.c	2010-11-18 18:58:48.000000000 +0100
+++ linux-2.6.37-rc2.new/kernel/sched.c	2010-11-18 19:00:55.000000000 +0100
@@ -78,6 +78,7 @@
 
 #include "sched_cpupri.h"
 #include "workqueue_sched.h"
+#include "sched_tty.h"
 
 #define CREATE_TRACE_POINTS
 #include <trace/events/sched.h>
@@ -615,11 +616,16 @@
  */
 static inline struct task_group *task_group(struct task_struct *p)
 {
+	struct task_group *tg;
 	struct cgroup_subsys_state *css;
 
 	css = task_subsys_state_check(p, cpu_cgroup_subsys_id,
 			lockdep_is_held(&task_rq(p)->lock));
-	return container_of(css, struct task_group, css);
+	tg = container_of(css, struct task_group, css);
+
+	tty_sched_check_attach(p, &tg);
+
+	return tg;
 }
 
 /* Change a task's cfs_rq and parent entity if it moves across CPUs/groups */
@@ -2016,6 +2022,7 @@
 #include "sched_idletask.c"
 #include "sched_fair.c"
 #include "sched_rt.c"
+#include "sched_tty.c"
 #include "sched_stoptask.c"
 #ifdef CONFIG_SCHED_DEBUG
 # include "sched_debug.c"
diff -Naur linux-2.6.37-rc2/kernel/sched_tty.h linux-2.6.37-rc2.new/kernel/sched_tty.h
--- linux-2.6.37-rc2/kernel/sched_tty.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.37-rc2.new/kernel/sched_tty.h	2010-11-18 19:04:21.000000000 +0100
@@ -0,0 +1,7 @@
+#ifdef CONFIG_SCHED_DESKTOP
+static inline void
+tty_sched_check_attach(struct task_struct *p, struct task_group **tg);
+#else
+static inline void
+tty_sched_check_attach(struct task_struct *p, struct task_group **tg) { }
+#endif
diff -Naur linux-2.6.37-rc2/kernel/sysctl.c linux-2.6.37-rc2.new/kernel/sysctl.c
--- linux-2.6.37-rc2/kernel/sysctl.c	2010-11-18 18:58:48.000000000 +0100
+++ linux-2.6.37-rc2.new/kernel/sysctl.c	2010-11-18 19:05:18.000000000 +0100
@@ -276,6 +276,17 @@
 		.mode		= 0644,
 		.proc_handler	= proc_dointvec,
 	},
+#ifdef CONFIG_SCHED_DESKTOP
+	{
+		.procname	= "sched_tty_sched_enabled",
+		.data		= &sysctl_sched_tty_sched_enabled,
+		.maxlen		= sizeof(unsigned int),
+		.mode		= 0644,
+		.proc_handler	= sched_tty_sched_handler,
+		.extra1		= &zero,
+		.extra2		= &one,
+	},
+#endif
 #ifdef CONFIG_SCHED_DEBUG
 	{
 		.procname	= "sched_min_granularity_ns",
