diff -ruEbBN aria2-1.10.9.old//src/OptionHandlerFactory.cc aria2-1.10.9.new/src/OptionHandlerFactory.cc
--- aria2-1.10.9.old//src/OptionHandlerFactory.cc	2011-01-10 09:12:21.000000000 +0100
+++ aria2-1.10.9.new/src/OptionHandlerFactory.cc	2011-01-27 04:49:40.000000000 +0100
@@ -352,8 +352,8 @@
     SharedHandle<OptionHandler> op(new NumberOptionHandler
                                    (PREF_MAX_CONNECTION_PER_SERVER,
                                     TEXT_MAX_CONNECTION_PER_SERVER,
-                                    "1",
-                                    1, 16,
+                                    "256",
+                                    1, 256,
                                     'x'));
     op->addTag(TAG_BASIC);
     op->addTag(TAG_FTP);
@@ -406,7 +406,7 @@
     SharedHandle<OptionHandler> op(new UnitNumberOptionHandler
                                    (PREF_MIN_SPLIT_SIZE,
                                     TEXT_MIN_SPLIT_SIZE,
-                                    "20M",
+                                    "1M",
                                     1024*1024, 1024*1024*1024,
                                     'k'));
     op->addTag(TAG_BASIC);
