diff -ruEbBN data.old/ext/fusefs_lib.c data.new/ext/fusefs_lib.c
--- data.old/ext/fusefs_lib.c	1970-01-01 01:00:00.000000000 +0100
+++ data.new/ext/fusefs_lib.c	2010-12-02 19:51:08.000000000 +0100
@@ -22,6 +22,25 @@
 #include <stdarg.h>
 #endif
 
+#if !defined(RUBY_VERSION_CODE) || \
+  RUBY_VERSION_CODE >= 190
+  
+#define STR2CSTR(x) rb_str2cstr(x, 0)
+char *
+rb_str2cstr(VALUE str,
+    long * len) {
+  StringValue(str);
+  if (len) {
+    *len = RSTRING_LEN(str);
+  } else if (RTEST(ruby_verbose) && RSTRING_LEN(str) != strlen(RSTRING_PTR(str))) {
+    rb_warn("string contains \\0 character");
+  }
+
+  return RSTRING_PTR(str);
+}
+
+#endif
+
 #include "fusefs_fuse.h"
 
 /* init_time
diff -ruEbBN data.old/lib/fusefs.rb data.new/lib/fusefs.rb
--- data.old/lib/fusefs.rb	1970-01-01 01:00:00.000000000 +0100
+++ data.new/lib/fusefs.rb	2010-12-02 19:52:16.000000000 +0100
@@ -4,7 +4,7 @@
 #
 # This includes helper functions, common uses, etc.
 
-require File.dirname(__FILE__) + '/../ext/fusefs_lib'
+require File.join(File.dirname(__FILE__), 'fusefs_lib')
 
 module FuseFS
   @running = true
